backend/core/views.py

```
@@ -5021,3 +5055,147 @@ def perform_destroy(self, instance):
+                )
+
+                # Replace the task dictionary with the actual TaskNode in the original list
+                tasks_list[i] = TaskNodeReadSerializer(task_node).data
+
+        return tasks_list
+
+    def perform_update(self, serializer):
+        task_template = serializer.save()
+
+        # Delete all existing TaskNode instances associated with this TaskTemplate
+        TaskNode.objects.filter(task_template=task_template).delete()
```

Ruff (0.8.2)

`TaskNode` may be undefined, or defined from star imports

(F405)